pg-index-health/src/main/java/io/github/mfvanek/pg/checks/host/BtreeIndexesOnArrayColumnsCheckOnHost.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2019-2024. Ivan Vakhrushev and others.
+ * https://github.com/mfvanek/pg-index-health
+ *
+ * This file is a part of "pg-index-health" - a Java library for
+ * analyzing and maintaining indexes health in PostgreSQL databases.
+ *
+ * Licensed under the Apache License 2.0
+ */
+
+package io.github.mfvanek.pg.checks.host;
+
+import io.github.mfvanek.pg.common.maintenance.Diagnostic;
+import io.github.mfvanek.pg.connection.PgConnection;
+import io.github.mfvanek.pg.model.PgContext;
+import io.github.mfvanek.pg.model.index.Index;
+
+import java.util.List;
+import javax.annotation.Nonnull;
+
+/**
+ * Check for B-tree indexes on array columns.
+ * GIN-index should be used instead for such columns.
+ *
+ * @author Vadim Khizhin
+ * @since 0.10.4
+ */
+public class BtreeIndexesOnArrayColumnsCheckOnHost extends AbstractCheckOnHost<Index> {
```

Comment:
I've added recently `IndexWithColumns` https://github.com/mfvanek/pg-index-health/blob/master/pg-index-health-model/src/main/java/io/github/mfvanek/pg/model/index/IndexWithColumns.java  
Please use it here.  
This is example https://github.com/mfvanek/pg-index-health/blob/328192f97307781c00fc6543a362e01d9c2f071c/pg-index-health/src/main/java/io/github/mfvanek/pg/checks/host/IndexesWithBooleanCheckOnHost.java#L51  
Also You need to tune your sql query in order to return index size
```
pg_relation_size(pi.indexrelid) as index_size,
```

Reply:
Done